File: backend/app/models/transcript_segment.py
from datetime import datetime

from sqlalchemy import (
    DateTime,
    Float,
    ForeignKey,
    Integer,
    Text,
)
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class TranscriptSegment(Base):

    __tablename__ = "transcript_segments"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        index=True
    )

    transcript_id: Mapped[int] = mapped_column(
        ForeignKey("transcripts.id"),
        nullable=False
    )

    segment_index: Mapped[int] = mapped_column(
        Integer,
        nullable=False
    )

    start_time: Mapped[float] = mapped_column(
        Float,
        nullable=False
    )

    end_time: Mapped[float] = mapped_column(
        Float,
        nullable=False
    )

    text: Mapped[str] = mapped_column(
        Text,
        nullable=False
    )

    # Whisper word timestamps stored as JSON
    words_json: Mapped[str | None] = mapped_column(
        Text,
        nullable=True
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow
    )

File: backend/Models/TranscriptSegment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VideoIntelligencePlatform.Backend.Models;

[Table("transcript_segments")]
public class TranscriptSegment
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("transcript_id")]
    public int TranscriptId { get; set; }

    [Column("segment_index")]
    public int SegmentIndex { get; set; }

    [Column("start_time")]
    public double StartTime { get; set; }

    [Column("end_time")]
    public double EndTime { get; set; }

    [Column("text")]
    public string Text { get; set; } = string.Empty;

    [Column("words_json")]
    public string? WordsJson { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey("TranscriptId")]
    public Transcript? Transcript { get; set; }
}
